### PR TITLE
network: disallow CNI networks with user namespaces

### PR DESCRIPTION
**Pull request by giuseppe**
it solves a segfault when running as rootless a command like:

$ podman run --uidmap 0:0:1 --net foo --rm fedora true
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x5629bccc407c]

goroutine 1 [running]:
panic(0x5629bd3d39e0, 0x5629be0ab8e0)
	/usr/lib/golang/src/runtime/panic.go:1064 +0x545 fp=0xc0004592c0 sp=0xc0004591f8 pc=0x5629bbd35d85
runtime.panicmem(...)
	/usr/lib/golang/src/runtime/panic.go:212
runtime.sigpanic()
	/usr/lib/golang/src/runtime/signal_unix.go:742 +0x413 fp=0xc0004592f0 sp=0xc0004592c0 pc=0x5629bbd4cd33
github.com/containers/podman/libpod.(*Runtime).setupRootlessNetNS(0xc0003fe9c0, 0xc0003d74a0, 0x0, 0x0)
	/builddir/build/BUILD/podman-2.2.1/_build/src/github.com/containers/podman/libpod/networking_linux.go:238 +0xdc fp=0xc000459338 sp=0xc0004592f0 pc=0x5629bccc407c
github.com/containers/podman/libpod.(*Container).completeNetworkSetup(0xc0003d74a0, 0x0, 0x0)
	/builddir/build/BUILD/podman-2.2.1/_build/src/github.com/containers/podman/libpod/container_internal.go:965 +0xb72 fp=0xc0004594d8 sp=0xc000459338 pc=0x5629bcc81732

[.....]

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: test/e2e/run_networking_test.go [hunk 1]
```diff
@@ -639,6 +639,22 @@ var _ = Describe("Podman run networking", func() {
 		Expect(create.ExitCode()).To(BeZero())
 	})
 
+	It("podman run in custom CNI network with --uidmap", func() {
+		netName := "podmantestnetwork2"
```
Comment (Luap99) [inline test/e2e/run_networking_test.go:643]: Please use `stringid.GenerateNonCryptoID()` for the network name. The static network names are causing a lot of flakes.

**Comment by rhatdan** [comment]
LGTM but make @Luap99 change.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Comment by mheon** [comment]
LGTM, but your test is failing on rootless.

**Comment by giuseppe** [comment]
it was passing locally as I had a modified containers.conf file :/

I don't think we can (easily) support this use case, the user namespace doesn't own the CNI namespace, so the netns configuration should be performed through the infra container after the container is created.

@AkihiroSuda do you have better ideas on how it can be performed?

**Comment by umohnani8** [comment]
LGTM, tests seem to be mainly green

**Comment by mheon** [comment]
/hold
One failure

**Comment by mheon** [comment]
Looks like a flake, already restarted. Remove hold when all green.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
All happy green test buttons.

**Pull request merged**